modules/ardrivo/test/String.cxx

```diff
@@ -0,0 +1,237 @@
+	REQUIRE_THAT(arr, EndsWith("H\x84jsan"));
+}
+
+TEST_CASE("Get a substring of a String" "[subsring]")
+{	
+	using Catch::Contains; 
+	String test = "test";
+	String newString = test.substring(2);
+	char* arr = new char[newString.length() + 1];
+	strcpy(arr, newString.c_str());
+	REQUIRE_THAT(arr, Contains("st"));
+}
+
+TEST_CASE("Copies the String�s characters to the supplied buffer" "[toCharArray]")
+{
+	using Catch::Contains; 
+	String test = "test";
+	char arr[5];
+	test.toCharArray(arr, test.length());
+	REQUIRE_THAT(arr, Contains("test"));
+}
+
+TEST_CASE("Converts a valid String to an integer. The input String should start with an integer number" "[toInt]") 
+{
+	String test = "1 and 23";
+	unsigned long num =  test.toInt();
+	REQUIRE(num == 1);
+	String test2 = "123";
+	unsigned long num2 =  test2.toInt();
+	REQUIRE(num2 == 123);
+}
+
+TEST_CASE("Converts a valid String to an double. The input String should start with an double number" "[toDouble]")
+{
+	String test = "1.0323414535 and 23";
+	double num =  test.toDouble();
+	REQUIRE(num == 1.0323414535);
+	String test2 = "123.4234512345";
+	double num2 =  test2.toDouble();
+	REQUIRE(num2 == 123.4234512345);
+}
+
+TEST_CASE("Converts a valid String to an float. The input String should start with an float number" "[toFloat]")
+{
+	String test = "1.0123 and 23";
+	float num =  test.toFloat();
+	REQUIRE(num == 1.0123f);
+	String test2 = "123.23545";
+	float num2 =  test2.toFloat();
+	REQUIRE(num2 == 123.23545f);
+}
+
+TEST_CASE("Get a lower-case version of a String" "[toLowerCase]")
+{
+	using Catch::Contains; 
+	String test = "I LIKE BANANAS";
+	test.toLowerCase();
+	char* arr = new char[test.length() + 1];
+	strcpy(arr, test.c_str());
+	REQUIRE_THAT(arr, Contains("i like bananas"));
+}
+
+TEST_CASE("Get a upper-case version of a String" "[toUpperCase]")
+{
+	using Catch::Contains; 
+	String test = "i like bigger bananas";
+	test.toUpperCase();
+	char* arr = new char[test.length() + 1];
+	strcpy(arr, test.c_str());
+	REQUIRE_THAT(arr, Contains("I LIKE BIGGER BANANAS"));
+}
+
+TEST_CASE("Remove spaces in a string" "[trim]")
+{
+	using Catch::Contains; 
+	String test = "i like bigger bananas";
+	test.trim();
+	char* arr = new char[test.length() + 1];
+	strcpy(arr, test.c_str());
+	REQUIRE_THAT(arr, Contains("ilikebiggerbananas"));
+}
+
+TEST_CASE("Checks if a string is equal to another" "[equals]")
+{
+	String test = "i like bigger bananas";
+	String test2 = "i LIKE BIGGER bananas";
+	bool isEqual = test.equals(test2);
+	REQUIRE(isEqual == false);
+}
+
+TEST_CASE("Checks if a string is equal to another and ignores case" "[equalsIgnoreCase]")
+{
+	String test = "i like bigger bananas";
+	String test2 = "i LIKE BIGGER bananas";
+	bool equalsIgnoreCase = test.equalsIgnoreCase(test2);
+	REQUIRE(equalsIgnoreCase == true);
+}
+
+TEST_CASE("Check operators" "[String::operator]")
+{
+	String test = "d";
+	String test2 = "AB";
+
+	SECTION("Operator ==")
+	{
+		REQUIRE(test.operator==(test2) == false);
```

Comment:
Why do you call the operator directly? also == false should be omitted, just use `!`.